data/src/main/java/com/annhienktuit/data/mappers/SongMapper.java

```diff
@@ -0,0 +1,34 @@
+package com.annhienktuit.data.mappers;
+
+import com.annhienktuit.data.models.SongData;
+import com.annhienktuit.domain.models.Song;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by Nhien Nguyen on 3/22/2022
+ */
+public class SongMapper {
```

Comment:
u should make a interface Mapper<Entity, Model> like this:

```
public interface Mapper<E, M> {

    E fromModel(M model);
    
    M fromEntity(E entity);
}
```
then implement it to your mapper.